safe-global/web-core
# fix: filter injected wallets per chain

## What it solves

Reimplementation of #703

## How this PR fixes it

Injected wallets are now adjusted according to whether they are present in the `disabledWallets` array.

All wallets are initially loaded and then [onboard's `setWalletModules` is called](https://github.com/safe-global/web-core/blob/main/src/hooks/wallets/useOnboard.ts#L131-L132) with the [supported wallets](https://github.com/safe-global/web-core/blob/main/src/hooks/wallets/wallets.ts#L81-L88). The `ChainInfo` is now passed to wallet modules to allow for [filtering](https://docs.blocknative.com/onboard/injected-wallets#excluding-wallets).

## How to test it

Open the Safe and view the available wallets. Disable an inject wallet (tally, frame, opera, trust or metamask) via the CGW and refresh the page. Observe the disabled wallet is no longer available.
